examples/prometheus-kubernetes/main.go

```diff
@@ -0,0 +1,1286 @@
+			"/etc/config/rules",
+			"/etc/config/alerts",
+		},
+		ScrapeConfigs: []*prometheus.ScrapeConfig{
+			{
+				JobName: "kubernetes-apiservers",
+				ServiceDiscoveryConfig: sdconfig.ServiceDiscoveryConfig{
+					KubernetesSDConfigs: []*kubernetes.SDConfig{
+						{
+							Role: kubernetes.RoleEndpoint,
+						},
+					},
+				},
+				RelabelConfigs: []*prometheus.RelabelConfig{
+					{
+						Action: prometheus.RelabelKeep,
+						Regex: prometheus.MustNewRegexp("default;kubernetes;https"),
+						SourceLabels: model.LabelNames{
+							"__meta_kubernetes_namespace",
+							"__meta_kubernetes_service_name",
+							"__meta_kubernetes_endpoint_port_name",
+						},
+					},
+				},
+				Scheme: "https",
+				HTTPClientConfig: config.HTTPClientConfig{
+					TLSConfig: config.TLSConfig{
+						CAFile: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+						InsecureSkipVerify: true,
+					},
+					BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+				},
+			},
+			{
+				JobName: "kubernetes-nodes",
+				ServiceDiscoveryConfig: sdconfig.ServiceDiscoveryConfig{
+					KubernetesSDConfigs: []*kubernetes.SDConfig{
+						{
+							Role: kubernetes.RoleNode,
+						},
+					},
+				},
+				RelabelConfigs: []*prometheus.RelabelConfig{
+					{
+						Action: prometheus.RelabelLabelMap,
+						Regex: prometheus.MustNewRegexp("__meta_kubernetes_node_label_(.+)"),
+					},
+					{
+						Replacement: "kubernetes.default.svc:443",
+						TargetLabel: "__address__",
+					},
+					{
+						Regex: prometheus.MustNewRegexp("(.+)"),
+						Replacement: "/api/v1/nodes/${1}/proxy/metrics",
+						TargetLabel: "__metrics_path__",
+						SourceLabels: model.LabelNames{
+							"__meta_kubernetes_node_name",
+						},
+					},
+				},
+				Scheme: "https",
+				HTTPClientConfig: config.HTTPClientConfig{
+					TLSConfig: config.TLSConfig{
+						CAFile: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+						InsecureSkipVerify: true,
+					},
+					BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+				},
+			},
+			{
+				JobName: "kubernetes-nodes-cadvisor",
+				ServiceDiscoveryConfig: sdconfig.ServiceDiscoveryConfig{
+					KubernetesSDConfigs: []*kubernetes.SDConfig{
+						{
+							Role: kubernetes.RoleNode,
+						},
+					},
+				},
+				RelabelConfigs: []*prometheus.RelabelConfig{
+					{
+						Action: prometheus.RelabelLabelMap,
+						Regex: prometheus.MustNewRegexp("__meta_kubernetes_node_label_(.+)"),
+					},
+					{
+						Replacement: "kubernetes.default.svc:443",
+						TargetLabel: "__address__",
+					},
+					{
+						Regex: prometheus.MustNewRegexp("(.+)"),
+						Replacement: "/api/v1/nodes/${1}/proxy/metrics/cadvisor",
+						TargetLabel: "__metrics_path__",
+						SourceLabels: model.LabelNames{
+							"__meta_kubernetes_node_name",
+						},
+					},
+				},
+				Scheme: "https",
+				HTTPClientConfig: config.HTTPClientConfig{
+					TLSConfig: config.TLSConfig{
```

Comment:
I think this is default, plus it repeats everywhere. Can we have helper here? (: